aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/service/AWSS3StorageService.kt

```diff
@@ -55,16 +55,27 @@ internal class AWSS3StorageService(
     private val awsRegion: String,
     private val s3BucketName: String,
     private val authCredentialsProvider: AuthCredentialsProvider,
-    private val awsS3StoragePluginKey: String
+    private val awsS3StoragePluginKey: String,
+    private val clientProvider: StorageTransferClientProvider
 ) : StorageService {
 
+    companion object {
+        @JvmStatic
+        fun getS3Client(region: String, authCredentialsProvider: AuthCredentialsProvider): S3Client {
```

Comment:
Should we just use the ClientProvider so that all s3 providers are created in same location. Also see comment about the credentialsProvider. I think we need to use this value for all created S3Clients.

Reply:
This is essentially part of the client provider delegated work/implementation.  It's the work-around to create an S3Client when region is the only field available and I can't create an S3Client in java code.

https://github.com/aws-amplify/amplify-android/pull/2899/files#diff-68c08b683554135bdca3ec8741ef1160af2a1add61a87d302af6719c214b46cdR152-R156

Let me know if there's a simpler way to do this.

Reply:
Alternatively, I can remove if `if(region != null)` block completely and always return the S3Client from the `defaultStorageService` to simplify the clientProvider delegate implementation.

Reply:
Refactored and move the static method to `S3StorageTransferClientProvider`